signer/tests/integration/postgres.rs

```diff
@@ -5881,6 +5881,199 @@ async fn is_withdrawal_active_for_considered_withdrawal() {
+        num_bitcoin_blocks: 30,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 0,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: 0,
+        consecutive_blocks: true,
+    };
+    // The number of signers does not matter
+    let num_signers = 1;
+    let signer_set = testing::wsts::generate_signer_set_public_keys(&mut rng, num_signers);
+    let test_data = TestData::generate(&mut rng, &signer_set, &test_model_params);
+    test_data.write_to(&mut db).await;
+
+    let context_window = 10;
+    let bitcoin_chain_tip = db
+        .get_bitcoin_canonical_chain_tip_ref()
+        .await
+        .unwrap()
+        .unwrap();
+    let current_total = db
+        .compute_withdrawn_total(&bitcoin_chain_tip.block_hash, context_window)
+        .await
+        .unwrap();
+    assert_eq!(current_total, 0);
+
+    // Let's add one withdrawal output into the chain tip of our blockchain
+    // and another output on a block at the same height that is not on our
+    // blockchain.
+    let amount1 = 123_456;
+    let output1 = model::TxOutput {
+        amount: amount1,
+        output_type: model::TxOutputType::Withdrawal,
+        ..Faker.fake_with_rng(&mut rng)
+    };
+
+    let tx = model::Transaction {
+        txid: output1.txid.into_bytes(),
+        tx: Vec::new(),
+        tx_type: model::TransactionType::SbtcTransaction,
+        block_hash: bitcoin_chain_tip.block_hash.into_bytes(),
+    };
+    db.write_bitcoin_transactions(vec![tx]).await.unwrap();
+    db.write_tx_output(&output1).await.unwrap();
+
+    // Okay, and another output on another blockchain
+    let amount2 = 789_101;
+    let another_block = model::BitcoinBlock {
+        block_height: bitcoin_chain_tip.block_height,
+        ..Faker.fake_with_rng(&mut rng)
+    };
+
+    let output2 = model::TxOutput {
+        amount: amount2,
+        output_type: model::TxOutputType::Withdrawal,
+        ..Faker.fake_with_rng(&mut rng)
+    };
+
+    let tx = model::Transaction {
+        txid: output2.txid.into_bytes(),
+        tx: Vec::new(),
+        tx_type: model::TransactionType::SbtcTransaction,
+        block_hash: another_block.block_hash.into_bytes(),
+    };
+    db.write_bitcoin_block(&another_block).await.unwrap();
+    db.write_bitcoin_transactions(vec![tx]).await.unwrap();
+    db.write_tx_output(&output2).await.unwrap();
+
+    // The blockchain with a context window of zero is just the chain_tip,
+    // same thing as a context window of 1. So this checks the "regular"
+    // chain tip.
+    let current_total = db
+        .compute_withdrawn_total(&bitcoin_chain_tip.block_hash, 0)
+        .await
+        .unwrap();
+
+    assert_eq!(current_total, amount1);
+
+    // Let's check for the withdrawal output on that other blockchain.
+    let current_total = db
+        .compute_withdrawn_total(&another_block.block_hash, 0)
+        .await
+        .unwrap();
+
+    assert_eq!(current_total, amount2);
```

Comment:
Can we also check that this holds for a longer context window (eg, 5)? To further test that this is not just a special case induced by the context_window = 0

Reply:
Yeah, no problem.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1503/commits/6e3282d6ac69fc9232d301f9d8cf1d79b9c7e223